### PR TITLE
- Added ability to answer surveys, as well minor edits to login funct…

### PR DESCRIPTION
**Pull request by Binyamin-Brion**
…ionality

### CODE REVIEW

**Review by frankliu197 — APPROVED** [review]

File: src/pages/AnswerSurvey.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+} from "@mui/material";
+import AddIcon from '@mui/icons-material/Add';
+import ArrowBackIcon from '@mui/icons-material/ArrowBack';
+
+import RestClient from '../utils/RestClient';
+import AppContext from '../contexts/AppContext';
+import NewQuestionDialog from '../components/NewQuestionDialog';
+import Question from '../components/Question';
+
+const EditSurvey = () => {
```
Comment (doodlehead) [inline src/pages/AnswerSurvey.js]: Rename

File: src/App.js [hunk 2]
```diff
@@ -61,6 +62,7 @@ function App() {
               <Routes>
                 <Route path="/survey" element={<SurveyList />} />
                 <Route path="/survey/:surveyId" element={<EditSurvey />} />
+                <Route path="/survey/:surveyId/progress" element={<AnswerSurvey/>}/>
```
Comment (doodlehead) [inline src/App.js:65]: Why is it named "progress"?

File: src/components/Question.js [hunk 3]
```diff
@@ -3,64 +3,140 @@ import {
...
+  MenuItem,
+  FormControl,
+  TextField
 } from "@mui/material";
 import DeleteIcon from '@mui/icons-material/Delete';
 import EditIcon from '@mui/icons-material/Edit';
 
-const Question = ({ question, onDelete, onEdit }) => {
+const Question = ({ question, display, onDelete, onEdit, onAnswer }) => {
+  if (display) {
```
Comment (doodlehead) [inline src/components/Question.js:17]: Not how you do this, move the "if" into the markup below

File: src/pages/SurveyList.js [hunk 4]
```diff
@@ -74,11 +78,9 @@ const SurveyList = () => {
           alignItems: 'center'
         }}>
         <Box sx={{ marginLeft: '12px' }}>{survey.id}.</Box>
-        <Box sx={{ flexGrow: 1 }}>
-          <Box>{survey.name}</Box>
-          <Typography>Number of questions: TODO?</Typography>
-          <Typography>Number of responses: TODO?</Typography>
-        </Box>
+        <Button sx={{ flexGrow: 1 }} onClick={()=> handleAnswerSurvey(survey.id)}>
+          {survey.name}
```
Comment (doodlehead) [inline src/pages/SurveyList.js:82]: This behavior is not obvious at all and doesn't really make sense from an end-user perspective

File: src/components/Question.js [hunk 5]
```diff
@@ -3,64 +3,140 @@ import {
...
+            onChange={e => onAnswer(e.target.value)}
+          >
+            {question.choices.map(q =>
+              <MenuItem value={q}>{q}</MenuItem>
+            )}
+          </Select>
+        </FormControl>
+
+      } else if (question.type === 'RANGE') {
+          const onRangeChange = e => {
```
Comment (doodlehead) [inline src/components/Question.js:88]: This doesn't work. Lots of buggy behavior.

File: src/components/Question.js [hunk 6]
```diff
@@ -3,64 +3,140 @@ import {
...
+      if (question.type === 'MC') {
 
-    } else if (question.type === 'RANGE') {
-      return <>
-        <Box>Min: {question.min}</Box>
-        <Box>Max: {question.max}</Box>
-      </>
-    } else {
-      return []
+        return <FormControl fullWidth>
```
Comment (doodlehead) [inline src/components/Question.js:75]: fullWidth looks pretty bad on Selects

File: src/pages/AnswerSurvey.js [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+  const appContext = useContext(AppContext);
+  const params = useParams();
+  const navigate = useNavigate();
+  const [survey, setSurvey] = useState(null)
+  const [answer, setAnswer] = useState([]);
+  useEffect(() => {
+    // Fetch the survey on load
+    fetchSurvey()
```
Comment (doodlehead) [inline src/pages/AnswerSurvey.js:25]: This won't work for users that are not authenticated.

**Review by doodlehead — CHANGES_REQUESTED** [review]
Fixes needed

File: src/pages/SurveyList.js (hunk 4 above)
Comment (Binyamin-Brion) [reply src/pages/SurveyList.js:82]: Not sure what the issue is here? When I go to answer surveys it looks fine to me

File: src/components/Question.js (hunk 5 above)
Comment (Binyamin-Brion) [reply src/components/Question.js:88]: What is the specific issue? It correctly handles incorrect and correct responses in terms of range

File: src/components/Question.js (hunk 6 above)
Comment (Binyamin-Brion) [reply src/components/Question.js:75]: Looks fine on my end?

File: src/components/Question.js (hunk 3 above)
Comment (Binyamin-Brion) [reply src/components/Question.js:17]: What is wrong with this method? It works as expected?

File: src/App.js (hunk 2 above)
Comment (Binyamin-Brion) [reply src/App.js:65]: What is wrong with progress? Users has made progress answering the survey

File: src/components/Question.js (hunk 6 above)
Comment (frankliu197) [reply src/components/Question.js:75]: Looks fine to me too

File: src/pages/AnswerSurvey.js (hunk 7 above)
Comment (Binyamin-Brion) [reply src/pages/AnswerSurvey.js:25]: We might want to revisit who needs  to be authenticated in the future. We can take a look later- not critical for showing being able to answer surveys

File: src/components/Question.js (hunk 5 above)
Comment (Binyamin-Brion) [reply src/components/Question.js:88]: There was some weird behavior when trying to remove digits, latest change fixes that

File: src/components/Question.js (hunk 3 above)
Comment (doodlehead) [reply src/components/Question.js:17]: It's poor code style but you don't have to change it if you don't want to...

File: src/pages/AnswerSurvey.js (hunk 7 above)
Comment (doodlehead) [reply src/pages/AnswerSurvey.js:25]: It *is* critical for answering surveys, the primary use case for answering surveys is unauthenticated users.

File: src/pages/SurveyList.js (hunk 4 above)
Comment (aeiou80) [reply src/pages/SurveyList.js:82]: Same here, looks fine

**Review by aeiou80 — APPROVED** [review]

File: src/pages/AnswerSurvey.js (hunk 7 above)
Comment (aeiou80) [reply src/pages/AnswerSurvey.js:25]: We can revisit in Milestone 3

Comment (Binyamin-Brion) [reply src/pages/AnswerSurvey.js:25]: Im going to add a backend change that allows non-authenticated users to answer a survey using a link to the survey

**Pull request merged**